rabbitmq: disable deprecated telegraf metrics

osnyx opened the pull request:
@flyingcircusio/release-managers

With the `management_metrics_collection` RabbitMQ feature being
deprecated, the native Telegraf plugin for RabbitMQ will stop working
and is being removed.
The rabbitmq_telegraf exporter serves as a replacement since
fc-nixos-25.05 and now becomes the sole metric source.

The corresponding Grafana dashboard PR is https://github.com/flyingcircusio/grafana/pull/22, but both can be merged independently.

PL-132502

## Release process

- [x] Created changelog in the upgrade docs

## PR release workflow (internal)

- [x] PR has internal ticket
- [x] internal issue ID (PL-…) part of branch name
- [x] internal issue ID mentioned in PR description text
- [x] ticket is on Platform agile board
- [x] ticket state set to *Pull request ready*
- [x] if ticket is more urgent than within the next few days, directly contact a member of the Platform team

- [ ] ensure all checks are green
- [ ] get a review from a colleague

## Design notes

- [x] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [x] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - [x] change management: provide transitioning path
    - properly timed to be released in a new major platform release
    - transitory period where new and deprecated metrics coexisted
    - proper announcement and documentation
    - migration: delete old monitoring user to avoid unnecessary user accounts
- [x] Security requirements tested? (EVIDENCE)
  - [x] automated tests still pass
  - [x] extended test coverage
  - [x] investigated metrics on a test VM, both via curl and via Grafana dashboard
  - [x] investigated telegraf config (and fixed a duplication flaw)
  - [x] tested clean up of old monitoring user on old and current system

osnyx committed:
telegraf: fix duplicate prometheus input definitions

Prometheus inputs defined via flyingcircus.services.telegraf.inputs were
accidentally added twice, once with a defined metric_version and once
without.

osnyx committed:
services/rabbitmq: drop deprecated telegraf exporter

With the `management_metrics_collection` RabbitMQ feature being
deprecated, the native Telegraf plugin for RabbitMQ will stop working
and is being removed.
The rabbitmq_telegraf exporter serves as a replacement since
fc-nixos-25.05 and now becomes the sole metric source.

PL-132502

osnyx committed:
tests/rabbitmq: chack for promettheus metrics exporting

For also testing the passthrough of the rabbitmq_prometheus exporter
through telegraf, this test deliberately re-enables telegraf.service.
PL-133391

Includes some workarounds for PL-134248.

osnyx commented:
The first commit might be worth being backported individually, although I have not seen it causing any issues in the wild so far.

ctheune commented on nixos/services/rabbitmq/default.nix:
```diff
@@ -284,26 +283,14 @@ in
         rabbitmqctl list_users | grep guest && \
           rabbitmqctl delete_user guest
 
-        # TODO: We can remove the fc-telegraf user when having migrated to prometheus only (PL-133391)
```
is the user being deleted somewhere?

osnyx replied:
Good catch, no.

osnyx replied:
Implemented.

osnyx committed:
services/rabbitmq: clean up old telegraf monitoring user

When migrating from 25.11 or older, there is a fc-telegraf monitoring
user to be cleaned up.
This is purely necessary for migration and may be reverted for the 26.11
release.

PL-133391
